safe_transaction_service/contracts/tx_decoder.py

```diff
@@ -552,6 +552,34 @@ class DbTxDecoder(TxDecoder):
 
     cache_abis_by_address = TTLCache(maxsize=2048, ttl=60 * 5)  # 5 minutes of caching
 
+    @classmethod
+    def get_most_relevant_contract_abi(cls, address: ChecksumAddress):
+        """
+        Retrieves the most relevant ABI for the contract at the given address.
+
+        :param address: Contract address
+        :return: List of ABI data if found, `None` otherwise.
+        """
+        return (
+            ContractAbi.objects.filter(contracts__address=address)
+            .order_by("relevance")
+            .values_list("abi", flat=True)
+            .first()
+        )
+
+    def contract_has_fallback_call(self, address: ChecksumAddress) -> bool:
+        """
+        :param address: Contract address
+        :return: `True` if the contract has a fallback call with `stateMutability` payable
+        """
+        abi = self.get_most_relevant_contract_abi(address)
+        if abi:
+            return any(
+                fn_abi.get("type") == "fallback"
+                and fn_abi.get("stateMutability") == "payable"
```

Comment:
Why must it be `payable`?

Reply:
You're right, my mistake! Fixed [here](https://github.com/safe-global/safe-transaction-service/pull/2262/commits/4caf91e5a242ebb07a6d3ff2cf11d225dbefc8da).